ttnn/cpp/ttnn/operations/moreh/moreh_layer_norm/device/moreh_layer_norm_device_operation.hpp

```diff
@@ -0,0 +1,84 @@
+
+namespace ttnn::operations::moreh::moreh_layer_norm {
+struct MorehLayerNormOperation {
+    struct operation_attributes_t {
+        uint32_t normalized_dims;
+        float eps;
+
+        const MemoryConfig memory_config;
+        const std::optional<DeviceComputeKernelConfig> compute_kernel_config;
+    };
+
+    struct tensor_args_t {
+        const Tensor& input;
+
+        const std::optional<const Tensor> gamma;
+        const std::optional<const Tensor> beta;
+        const std::optional<const Tensor> output;
+        const std::optional<const Tensor> mean;
+        const std::optional<const Tensor> rstd;
+    };
+
+    using shape_return_value_t = std::vector<Shape>;
+    using tensor_return_value_t = std::vector<std::optional<Tensor>>;
+
+    struct ProgramFactory {
+        struct shared_variables_t {
+            KernelHandle unary_reader_kernel_id;
+            KernelHandle unary_writer_kernel_id;
+            std::size_t num_cores;
+            std::size_t num_cores_y;
+        };
+
+        using cached_program_t = ttnn::device_operation::CachedProgram<shared_variables_t>;
+
+        static cached_program_t create(
+            const operation_attributes_t& operation_attributes,
+            const tensor_args_t& tensor_args,
+            tensor_return_value_t& output_tensor);
+
+        static void override_runtime_arguments(
+            cached_program_t& cached_program,
+            const operation_attributes_t& operation_attributes,
+            const tensor_args_t& tensor_args,
+            tensor_return_value_t& output_tensor);
+    };
+
+    using program_factory_t = std::variant<ProgramFactory>;
+
+    static void validate_inputs(const operation_attributes_t&, const tensor_args_t&);
+    static program_factory_t select_program_factory(const operation_attributes_t&, const tensor_args_t&);
+    static void validate_on_program_cache_miss(const operation_attributes_t&, const tensor_args_t&);
+    static void validate_on_program_cache_hit(const operation_attributes_t&, const tensor_args_t&);
+    static shape_return_value_t compute_output_shapes(const operation_attributes_t&, const tensor_args_t&);
+    static tensor_return_value_t create_output_tensors(const operation_attributes_t&, const tensor_args_t&);
+    static std::tuple<operation_attributes_t, tensor_args_t> invoke(
+        const Tensor& input,
+        const uint32_t normalized_dims,
+        const float eps,
+        const std::optional<const Tensor> gamma,
+        const std::optional<const Tensor> beta,
+        const std::optional<const Tensor> output,
+        const std::optional<const Tensor> mean,
+        const std::optional<const Tensor> rstd,
+        const std::optional<MemoryConfig>& memory_config,
+        const std::optional<DeviceComputeKernelConfig>& compute_kernel_config);
+};
+}  // namespace ttnn::operations::moreh::moreh_layer_norm
+
+namespace ttnn::prim {
+constexpr auto moreh_layer_norm = ttnn::register_operation_with_auto_launch_op<
```

Comment:
```suggestion
constexpr auto moreh_layer_norm = ttnn::register_operation<
```
Please, use `register_operation` for prim namespace

Reply:
This is the final comment. Once this is fixed, I'll approve it.

Reply:
Sorry for not noticing this. I’ve fixed it.